lib/path.rb

```diff
@@ -0,0 +1,76 @@
+class Metis
+    class Path
+      # Assumes Metis file paths are in the form
+      #    metis://<project>/<bucket>/<folder path>/<file name>
+      # Splitting the above produces
+      #   ["metis", "", "<project>", "<bucket>", "<folder path>" ... "file name"]
+
+      FILENAME_MATCH=/[^<>:;,?"*\|\/\x00-\x1f]+/x
+
+      FILEPATH_MATCH=%r!
+        \A
+          metis:\/\/
+          (?<project_name>(#{FILENAME_MATCH.source}))
+          /
+          (?<bucket_name>(#{FILENAME_MATCH.source}))
+          /
```

Comment:
I think both project_name and bucket_name are only \w+; also I worry that FILENAME_MATCH here will diverge from the one in Metis::File. I guess you can't build the constant using Metis::File::FILENAME_MATCH because of load order issues? Alternatively you could build it on-demand in a memoized class method (like `def self.filepath_match; @filepath_match ||= etc.; end`)

Reply:
That's a great point, especially if we decide to loosen up the restrictions on the filename. Load order does prevent just using `Metis::File::FILENAME_MATCH` directly, so I've changed it to a memoized class method.